### PR TITLE
MODAUD-21 - Notices: log actions

### PR DESCRIPTION
**Pull request by siarhei-charniak**
[MODAUD-21](https://issues.folio.org/browse/MODAUD-21) - Notices: log actions

## Purpose
Log events for Notices

## Approach
Minor improvements to collect data for [Circulation Audit Logs](https://wiki.folio.org/display/DD/Circulation+Audit+logs)

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by marcjohnson-kint — COMMENTED** [review]
Changes to a module should be associated with a JIRA issue in the relevant project for the module. Please raise an issue within the circulation project and associate these changes with that issue.

File: src/main/java/org/folio/circulation/domain/notice/PatronNoticeService.java [hunk 1]
```diff
@@ -57,7 +61,9 @@ public PatronNoticeService(PatronNoticePolicyRepository noticePolicyRepository,
     patronNotice.setOutputFormat(noticeConfig.getFormat().getOutputFormat());
     patronNotice.setContext(context);
 
-    return sendNotice(patronNotice);
+    return noticePolicyRepository.lookupPolicyId(loan.getItem(), loan.getUser())
+      .thenAccept(result -> patronNotice.getContext().put(NOTICE_POLICY_ID, result.value().getPolicyId()))
```
Comment (bohdan-suprun) [inline src/main/java/org/folio/circulation/domain/notice/PatronNoticeService.java:65]:
```suggestion
      .thenApply(result -> result.map(policy->  patronNotice.getContext().put(NOTICE_POLICY_ID, policy.getPolicyId()))
```

File: src/main/java/org/folio/circulation/domain/notice/PatronNoticeService.java [hunk 2]
```diff
@@ -57,7 +61,9 @@ public PatronNoticeService(PatronNoticePolicyRepository noticePolicyRepository,
...
+      .thenCompose(vVoid -> sendNotice(patronNotice));
```
Comment (bohdan-suprun) [inline src/main/java/org/folio/circulation/domain/notice/PatronNoticeService.java:66]:
```suggestion
      .thenCompose(result-> result.after(vVoid -> sendNotice(patronNotice)))
```

File: src/main/java/org/folio/circulation/domain/notice/PatronNoticeService.java [hunk 3]
```diff
@@ -71,7 +77,10 @@ public PatronNoticeService(PatronNoticePolicyRepository noticePolicyRepository,
 
   private CompletableFuture<Result<Pair<PatronNoticeEvent, String>>> loadNoticePolicyId(PatronNoticeEvent event) {
     return noticePolicyRepository.lookupPolicyId(event.getItem(), event.getUser())
-      .thenApply(mapResult(circulationRuleMatchEntity -> Pair.of(event, circulationRuleMatchEntity.getPolicyId())));
+      .thenApply(mapResult(circulationRuleMatchEntity -> {
+        event.getNoticeContext().put(NOTICE_POLICY_ID, circulationRuleMatchEntity.getPolicyId());
```
Comment (bohdan-suprun) [inline src/main/java/org/folio/circulation/domain/notice/PatronNoticeService.java:81]: Can this side effect will be done after receiving response from the method?

Alternatively,  we can reflect the side effect in method name, e.g. - `loadNoticePolicyIdAndPutInContext`

**Comment by bohdan-suprun** [comment]
Hi @alexanderkurash, @OleksandrVidinieiev, @roman-barannyk,

Could you please take a look, as the changes in patron notices logic. Thanks!

**Pull request closed without merging**